domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/ActivityLifecycleObserver.kt

```diff
@@ -0,0 +1,79 @@
+package org.oppia.android.domain.oppialogger.analytics
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import org.oppia.android.domain.oppialogger.ApplicationStartupListener
+import org.oppia.android.util.system.OppiaClock
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val APPLICATION_STARTUP_SCREEN = "application_startup_screen"
+
+/** Observer that observes activity lifecycle. */
+@Singleton
+class ActivityLifecycleObserver @Inject constructor(
+  private val oppiaClock: OppiaClock,
+  private val application: Application,
+  private val performanceMetricsLogger: PerformanceMetricsLogger
+) : Application.ActivityLifecycleCallbacks, ApplicationStartupListener {
+
+  /**
+   * Timestamp indicating the time of application start-up. It will be used to calculate the
+   * cold-startup latency of the application.
+   *
+   * We're using a large Long value such that the time difference based on any timestamp will be
+   * negative and thus ignored until the app records initial time during [onCreate].
+   */
+  private var initialTimestamp: Long = Long.MAX_VALUE
+
+  /**
+   * Returns a boolean flag that makes sure that startup latency is logged only once in the entire
+   * application lifecycle.
+   * */
+  private var isStartupLatencyLogged: Boolean = false
+
+  private var currentScreen: String? = null
+
+  /**
+   * Returns the current active UI screen that's visible to the user. If UI is in-active or the app
+   * is in background, this value returns null.
+   */
+  fun getCurrentScreen(): String? = currentScreen
+
+  override fun onCreate() {
+    initialTimestamp = oppiaClock.getCurrentTimeMs()
+    application.registerActivityLifecycleCallbacks(this)
+    performanceMetricsLogger.logApkSize(APPLICATION_STARTUP_SCREEN)
+    performanceMetricsLogger.logStorageUsage(APPLICATION_STARTUP_SCREEN)
+  }
+
+  override fun onActivityResumed(activity: Activity) {
+    currentScreen = activity.javaClass.simpleName
+    if (!isStartupLatencyLogged) {
+      performanceMetricsLogger.logStartupLatency(
+        initialTimestamp,
```

Comment:
Should we maybe be logging the delta (i.e. the number of milliseconds to startup) instead of the timestamp?

Reply:
The delta was actually getting logged in the logger directly, however I think that might lead to less-accurate readings than the ones we'll get from the observer. Moved delta calculation here.